devops/providers/azure-devops/templates/infrastructure/cicd/azure-pipelines.yaml

```diff
@@ -0,0 +1,125 @@
+variables:
+
+# location of files used by pipeline
+- name: PIPELINE_SCRIPTS_ROOT
+  value: cicd
+- name: BUILD_DIR
+  value: build
+
+# dependency versions
+- name: GO_VERSION
+  value: '1.12.5'
+- name: TF_VERSION
+  value: '0.12.2'
+
+# terraform flags
+- name: TF_WARN_OUTPUT_ERRORS
+  value: 1
+- name: ARM_PROVIDER_STRICT
+  value: 'true'
+
+# test harness image location and image tagging metadata
+- name: TEST_HARNESS_BASE_IMAGE
+  value: 'msftcse/cobalt-test-base'
+- name: TEST_HARNESS_BUILD_ARTIFACT_IMAGE_PREFIX
+  value: cobalt-build-artifact
+- name: TEST_HARNESS_REPOSITORY_GIT_URL
+  value: https://github.com/microsoft/cobalt.git
+- name: TEST_HARNESS_DIR
+  value: test-harness
+
+
+stages:
+- stage: CI
+  jobs:
+
+  - job: CI
+
+
+    ### BEGIN BUILD TASKS
+
+    pool: $(AGENT_POOL)
+    workspace:
+      clean: all
+
+    steps:
+    - checkout: self
+      clean: all
+      fetchDepth: 1
+
+    - script: bash ./$(PIPELINE_SCRIPTS_ROOT)/test-harness-checkout.sh
```

Comment:
Why do we need to checkout a second time? What is the purpose of the `checkout: self `in line 46?

Reply:
Only 1 checkout now